7a_temp_coop_munge/src/data_parsers/parse_UniversityofMissouri_LimnoProfiles_2017_2020.R

```diff
@@ -0,0 +1,203 @@
+      Timezone = NA,
+      depth = depth,
+      temp = temp,
+      Missouri_ID = lake_id,
+      .keep = 'none'
+    )
+
+  # combine all data sets together ------------------------------------------
+  data_clean <- dplyr::bind_rows(
+    dat_2017,
+    dat_2018_2020,
+    all_hw_files
+  )
+
+  saveRDS(object = data_clean, file = outfile)
+  sc_indicate(ind_file = outind, data_file = outfile)
+
+}
+
+# Read data functions -----------------------------------
+
+#' Read and clean files from 2019-2020 UMO data
+#'
+#' @param full_path chr, full file path
+#' @param keep_cols chr, vector of column names to keep
+#'
+read_subset_consistent_data <- function(full_path, keep_cols) {
+  dat <- suppressMessages(readr::read_csv(full_path, name_repair = 'universal') )
+
+  # convert field names to lowercase because field names are consistent but
+  # use of capitalization is not
+  names(dat) <- tolower(names(dat))
+  dat <- dat %>% dplyr::select(all_of(tolower(keep_cols)))
+
+  # extract lake id from file name because
+  # use of `Site.Name` field is inconsistent
+  dat$lake_id <- paste('Missouri_', extract_lake_id(full_path), sep = '')
+
+  return(dat)
+}
+
+#' Read and clean hand-transcribed files from 2017 and 2018 UMO data
+#'
+#' this function handles all files tagged `*_HW*`except `092_08_02_2017 HW.csv`.
+#'
+#' @param full_path chr, full file path
+#' @param skip_cols num, number of columns to skip before starting to read data.frame
+#' @param skip_rows num, number of rows to skip before starting to read data.frame
+#' @param depth_position num, column number associated with `depth`
+#' @param temp_position num, column number associated with `temperature`
+#'
+read_subset_hw_files <- function(full_path, skip_cols = 0, skip_rows = 2,
+                                 depth_position = 1, temp_position = 2) {
+  rstart <- skip_rows + 1
+  cstart <- skip_cols + 1
+
+  dat <- suppressMessages(read_csv(full_path, col_names = F)) %>%
+    remove_empty_columns()
+
+  col_nms <- as.character(dat[rstart, ]) %>%
+    tolower %>%
+    .[cstart:length(.)]
+
+  dat <- dat %>% .[(rstart + 1):nrow(.), cstart:ncol(.)]
+  names(dat) <- col_nms
+
+  dat <- dat %>%
+    dplyr::mutate(
+      depth = as.numeric(.[[(depth_position - skip_cols)]]),
+      temp = as.numeric(.[[(temp_position - skip_cols)]]),
+      date = extract_umo_date(full_path),
+      time = NA, # skipping time for now, in the manual files the time moves around
+      lake_id = paste('Missouri_', extract_lake_id(full_path), sep = '')
+    ) %>%
+    dplyr::select(depth, temp, date, time, lake_id) # name of temp and depth vary
+
+  return(dat)
+}
+
+# Additional helper functions ----------------------------------
+
+#' Extract the UMO Lake ID from the file name
+#'
+#' The lake ID is not always included in the data. This function creates
+#' a lake ID based on the info provided in the file name.
+#'
+#' @param file_path chr, full file path
+#'
+extract_lake_id <- function(full_path) {
```

Comment:
I think this function name may be too generic considering that it is designed perhaps to cover the UMO files only. Perhaps call it `extract_UMO_lake_id()` instead?

Reply:
Will update